### PR TITLE
Update: Add the ability to change api group and api version via a build arg

### PR DESCRIPTION
**Pull request by mxpaspa**
related to issue: https://github.com/razee-io/Razee/issues/100

**Changes**:
- added env vars for kube api group and version
- added references to api group, version in index.js
- added references to api group, version in resource.yaml

### CODE REVIEW

**Comment by mxpaspa** [comment]
> is this something you want reviewed yet?

Not yet, It should be ready tomorrow.

File: .travis.yml [hunk 1]
```diff
@@ -5,13 +5,16 @@ node_js:
...
+  global:
+    - GROUP=deploy.razee.io
+    - VERSION=v1alpha2
 script:
   # Audit npm packages. Fail build whan a PR audit fails, otherwise report the vulnerability and proceed.
   - if [ "${TRAVIS_PULL_REQUEST}" != "false" ]; then npx audit-ci --low; else npx audit-ci --low || true; fi
   - npm run lint
   - npm test
-  - docker build --rm -t "quay.io/razee/remoteresources3:${TRAVIS_COMMIT}" .
+  - docker build --rm -t "quay.io/razee/remoteresources3:${TRAVIS_COMMIT}" --build-arg GROUP=${GROUP} --build-arg VERSION=${VERSION} .
```
Comment (alewitt2) [inline .travis.yml:17]: is it intentional that the build-arg defined here is `GROUP` and `VERSION` but in the dockerfile is `GROUP_ARG` and `VERSION_ARG`? i thought these had to match.

File: kubernetes/RemoteResourceS3/resource.yaml [hunk 2]
```diff
@@ -74,16 +74,16 @@ items:
...
         razee.io/git-repo: "{{{GIT_REMOTE}}}"
         razee.io/commit-sha: "{{TRAVIS_COMMIT}}"
     spec:
       # group name to use for REST API: /apis/<group>/<version>
-      group: deploy.razee.io
+      group: "{{GROUP}}"
       # list of versions supported by this CustomResourceDefinition
       versions:
-        - name: v1alpha1
+        - name: "{{VERSION}}"
```
Comment (alewitt2) [inline kubernetes/RemoteResourceS3/resource.yaml:86]: you are overwriting `v1alpha1`, but you should be overwriting `v1alpha2` line 147.

File: .travis.yml (hunk 1 above)
Comment (mxpaspa) [reply .travis.yml:17]: yes you're right

Comment (alewitt2) [reply .travis.yml:17]: im right that they should match, or that it is intentional?

Comment (mxpaspa) [reply .travis.yml:17]: ha sorry, yes they should match. That was a last minute change.

File: .travis.yml [hunk 3]
```diff
@@ -5,13 +5,16 @@ node_js:
...
+  global:
+    - GROUP=deploy.razee.io
+    - VERSION=v1alpha2
 script:
   # Audit npm packages. Fail build whan a PR audit fails, otherwise report the vulnerability and proceed.
   - if [ "${TRAVIS_PULL_REQUEST}" != "false" ]; then npx audit-ci --low; else npx audit-ci --low || true; fi
   - npm run lint
   - npm test
-  - docker build --rm -t "quay.io/razee/remoteresources3:${TRAVIS_COMMIT}" .
+  - docker build --rm -t "quay.io/razee/remoteresources3:${TRAVIS_COMMIT}" --build-arg GROUP=${GROUP_ARG} --build-arg VERSION=${VERSION_ARG} .
```
Comment (alewitt2) [inline .travis.yml:17]:
```suggestion
  - docker build --rm -t "quay.io/razee/remoteresources3:${TRAVIS_COMMIT}" --build-arg GROUP_ARG=${GROUP} --build-arg VERSION_ARG=${VERSION} .
```
i believe you meant to do this?

**Pull request merged**